llama-index-integrations/vector_stores/llama-index-vector-stores-qdrant/llama_index/vector_stores/qdrant/base.py

```diff
@@ -654,17 +653,11 @@ async def _acreate_collection(self, collection_name: str, vector_size: int) -> N
 
     def _collection_exists(self, collection_name: str) -> bool:
         """Check if a collection exists."""
-        try:
-            return self._client.collection_exists(collection_name)
-        except (RpcError, UnexpectedResponse, ValueError):
-            return False
+        return self._client.collection_exists(collection_name)
 
     async def _acollection_exists(self, collection_name: str) -> bool:
         """Asynchronous method to check if a collection exists."""
-        try:
-            return await self._aclient.collection_exists(collection_name)
-        except (RpcError, UnexpectedResponse, ValueError):
-            return False
+        return await self._aclient.collection_exists(collection_name)
```

Comment:
Was there a specific problem this was causing? I'm pretty sure we had to wrap with a try except for a reason (but I'm forgetting the exact reason now 😅 )

Reply:
There were users who got timeouts(`UnexpectedResponse`) on the `collection_exists` calls. Upon which, the current implementation returned `False`. Leading to their collection being recreated, even though it existed.

Reply:
Ok, looking back, previously the try/except was there because we were using get_collection instead of collection_exists

Then when we switched methods, we kept the try except

Assuming collection_exists is reliable, and not prone to race conditions, this should be ok